src/shared/components/common/language-select.tsx

```diff
@@ -144,8 +146,95 @@ export class LanguageSelect extends Component<LanguageSelectProps, any> {
     i.props.onChange(selected);
   }
 
-  handleDeselectAll(i: LanguageSelect, event: any) {
-    event.preventDefault();
-    i.props.onChange([]);
+  get multiSelectBtn() {
+    const selectedLangs = this.props.selectedLanguageIds ?? [];
+    const filteredLangs = selectableLanguages(
+      this.props.allLanguages,
+      this.props.siteLanguages,
+      this.props.showAll,
+      this.props.showSite,
+      this.props.myUserInfo,
+    )
+      // Apply the filter
+      .filter(a =>
+        a.name.toLowerCase().includes(this.state.filter.toLowerCase()),
+      )
+      // Show selected items first
+      .sort((_, b) => {
+        if (selectedLangs.includes(b.id)) {
+          return 1;
+        } else {
+          return 0;
+        }
+      });
+
+    return (
+      <div className="border rounded w-100 bg-body">
+        <input
+          name="q"
+          type="search"
+          className="form-control flex-initial  border border-0"
+          placeholder={`${I18NextService.i18n.t("search")}...`}
+          aria-label={I18NextService.i18n.t("search")}
+          onInput={linkEvent(this, this.handleSearchChange)}
+          minLength={1}
+          value={this.state.filter}
+          autoComplete="off"
+        />
+        <hr className="border-dark border-2 m-0" />
+        <ul
+          class="list-group overflow-scroll"
+          style="max-height: 200px;"
+          tabIndex={-1}
+        >
+          {!this.props.multiple && (
+            <option selected disabled hidden>
+              {I18NextService.i18n.t("language_select_placeholder")}
+            </option>
+          )}
+          {filteredLangs.map(l => (
+            <li
+              className="list-group-item m-1 p-0 focus-ring border border-0 rounded-0"
+              id={l.id.toString()}
+              tabIndex={0}
+              onClick={linkEvent(this, this.handleCheckboxLanguageChange)}
+              onKeyPress={linkEvent(this, this.handleCheckboxLanguageChange)}
+              role="checkbox"
+              aria-checked={selectedLangs?.includes(l.id)}
```

Comment:
This throws a lint error:
```
/home/felix/workspace/lemmy/lemmy-ui/src/shared/components/common/language-select.tsx
202:15  error  Non-interactive elements should not be assigned interactive roles  jsx-a11y/no-noninteractive-element-to-interactive-role
```

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-noninteractive-element-to-interactive-role.md

Should I add a `<button>` element inside of `<li>` to fix this?